orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.cc

```diff
@@ -309,7 +309,8 @@ void IterateSubgraphFromNode(Graph& graph,
         continue;
       }
       auto func_name = static_cast<std::string>(cur->GetAttributes().at("name").s());
-      if (func_name == "_InspectActivation" || func_name == "_IncrementStep") {
+      if (func_name == "onnxruntime.training.utils.hooks._subscriber_manager._InspectActivation" ||
+          func_name == "onnxruntime.training.utils.hooks._subscriber_manager._IncrementStep") {
```

Comment:
`"onnxruntime.training.utils.hooks._subscriber_manager._InspectActivation"` and `"onnxruntime.training.utils.hooks._subscriber_manager._IncrementStep"` are used over and over. maybe a constant to prevent typos that would cause bugs?

Reply:
The concern makes senses. I defined constant std::string for now locally as a temporary fix, the longer term fix is: we had a work to allow defining customized shape inferencing from Python binding APIs, then we won't have such named based check any more later.